python/test.py

```diff
@@ -161,6 +161,19 @@ def test_enumeration_values(self):
     def test_select_all(self):
         self.assertEqual(self.test_obj.select_all().flat_size, 6)
 
+    def test_match_values(self):
+        # string
+        self.assertEqual(self.test_obj.match_values("attr-Z", "bb").flatten().tolist(),
+                         [1])
+
+        # enum
+        self.assertEqual(self.test_obj.match_values("E-mapping-good", "C").flatten().tolist(),
+                         [0, 2, 4, 5])
+
+        # numeric
+        for type_ in (float, int, ):
+            self.assertEqual(self.test_obj.match_values("attr-Y", type_(23)).flatten().tolist(),
+                             [2, ])
 
```

Comment:
Test it also with a float with many digits

Reply:
Since it's an exact comparison, I'm not sure what that is testing, could you elaborate as to why?

Reply:
Because using an int as a float is only testing an edge case. And also because I don't know what would happen if you would save an irrational number on disk (let's say, `sqrt(2)`) and try to find it back afterwards.

Reply:
It's just bits, so it would compare equal if the bits match, exactly how an represent-able integer gets compared.  Ie; double(sqrt(2.)) is (represented as) `0xcd3b7f669ea0f63f`, and double(2) is `0x0000000000000040`.